Content.Shared/Administration/SharedNetworkResourceManager.cs

```diff
@@ -0,0 +1,83 @@
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Robust.Shared.ContentPack;
+using Robust.Shared.Network;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Administration;
+
+/// <summary>
+///     Manager that allows resources to be added at runtime by admins.
+///     They will be sent to all clients automatically.
+/// </summary>
+public abstract class SharedNetworkResourceManager : IContentRoot
```

Comment:
I really do not like that this manager directly implements `IContentRoot`. Makes for very messy object lifetimes.

Reply:
I think I'll move this `IContentRoot` implementation to engine, given it's generic enough that multiple things or games may want to make use of it.

Reply:
I've done that now in https://github.com/space-wizards/RobustToolbox/pull/2590, is it better now?